### PR TITLE
Diego flores

### PR DESCRIPTION
**Pull request by Dfloresa**
Spring JPA, ExceptionHandler

### CODE REVIEW

File: DiegoFlores/SpringCRUD/src/main/java/com/bootcamp/vetPatitas/SpringCrudApplication.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package com.bootcamp.vetPatitas;
```
Comment (luisod78) [inline DiegoFlores/SpringCRUD/src/main/java/com/bootcamp/vetPatitas/SpringCrudApplication.java:1]: https://google.github.io/styleguide/javaguide.html#s5.2.1-package-names

File: DiegoFlores/SpringCRUD/src/main/java/com/bootcamp/vetPatitas/config/ExceptionManager.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
+package com.bootcamp.vetPatitas.config;
```
Comment (luisod78) [inline DiegoFlores/SpringCRUD/src/main/java/com/bootcamp/vetPatitas/config/ExceptionManager.java:1]: https://google.github.io/styleguide/javaguide.html#s5.2.1-package-names

File: DiegoFlores/SpringCRUD/src/main/java/com/bootcamp/vetPatitas/dao/PetDAO.java [hunk 3]
```diff
@@ -0,0 +1,9 @@
+package com.bootcamp.vetPatitas.dao;
```
Comment (luisod78) [inline DiegoFlores/SpringCRUD/src/main/java/com/bootcamp/vetPatitas/dao/PetDAO.java:1]: https://google.github.io/styleguide/javaguide.html#s5.2.1-package-names

File: DiegoFlores/SpringCRUD/src/main/java/com/bootcamp/vetPatitas/dto/PetDTO.java [hunk 4]
```diff
@@ -0,0 +1,69 @@
+package com.bootcamp.vetPatitas.dto;
```
Comment (luisod78) [inline DiegoFlores/SpringCRUD/src/main/java/com/bootcamp/vetPatitas/dto/PetDTO.java:1]: https://google.github.io/styleguide/javaguide.html#s5.2.1-package-names

File: DiegoFlores/SpringCRUD/src/main/java/com/bootcamp/vetPatitas/dto/ResponseDTO.java [hunk 5]
```diff
@@ -0,0 +1,40 @@
+package com.bootcamp.vetPatitas.dto;
```
Comment (luisod78) [inline DiegoFlores/SpringCRUD/src/main/java/com/bootcamp/vetPatitas/dto/ResponseDTO.java:1]: https://google.github.io/styleguide/javaguide.html#s5.2.1-package-names

File: DiegoFlores/SpringCRUD/src/main/java/com/bootcamp/vetPatitas/entity/Pet.java [hunk 6]
```diff
@@ -0,0 +1,80 @@
+package com.bootcamp.vetPatitas.entity;
```
Comment (luisod78) [inline DiegoFlores/SpringCRUD/src/main/java/com/bootcamp/vetPatitas/entity/Pet.java:1]: https://google.github.io/styleguide/javaguide.html#s5.2.1-package-names

File: DiegoFlores/SpringCRUD/src/main/java/com/bootcamp/vetPatitas/service/PetServiceImp.java [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+import com.bootcamp.vetPatitas.dto.PetDTO;
+import com.bootcamp.vetPatitas.exception.InvalidDataException;
+import com.bootcamp.vetPatitas.mapper.PetMapper;
+
+
+@Service
+public class PetServiceImp implements PetService{
+
+	@Autowired
+	PetMapper petMapper;
```
Comment (luisod78) [inline DiegoFlores/SpringCRUD/src/main/java/com/bootcamp/vetPatitas/service/PetServiceImp.java:22]: modifiers

File: DiegoFlores/SpringCRUD/src/main/java/com/bootcamp/vetPatitas/service/PetServiceImp.java [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+	
+	@Autowired
+	PetDAO petDao;
+	
+	@Override
+	public PetDTO addPet(PetDTO petDTO) {
```
Comment (luisod78) [inline DiegoFlores/SpringCRUD/src/main/java/com/bootcamp/vetPatitas/service/PetServiceImp.java:28]: great!! you validate all input data

PS. Maybe you would like to read:
https://www.baeldung.com/javax-validation

File: DiegoFlores/SpringCRUD/src/main/java/com/bootcamp/vetPatitas/userController/PetController.java [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+import com.bootcamp.vetPatitas.utils.Constants.ResponseConstants;
+
+@RestController
+@RequestMapping("pets")
+public class PetController {
+
+	@Autowired
+	PetService petService;
+	
+	@PutMapping(path = "/savePet",produces = MediaType.APPLICATION_JSON_VALUE)
```
Comment (luisod78) [inline DiegoFlores/SpringCRUD/src/main/java/com/bootcamp/vetPatitas/userController/PetController.java:31]: Prefer nouns to name resources (path mapping)
https://restfulapi.net/resource-naming/

File: DiegoFlores/SpringCRUD/src/test/java/com/bootcamp/vetPatitas/SpringCrudApplicationTests.java [hunk 10]
```diff
@@ -0,0 +1,13 @@
+package com.bootcamp.vetPatitas;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.boot.test.context.SpringBootTest;
+
+@SpringBootTest
+class SpringCrudApplicationTests {
+
+	@Test
+	void contextLoads() {
```
Comment (luisod78) [inline DiegoFlores/SpringCRUD/src/test/java/com/bootcamp/vetPatitas/SpringCrudApplicationTests.java:10]: a test without an assert?

File: DiegoFlores/SpringCRUD/src/main/java/com/bootcamp/vetPatitas/service/PetServiceImp.java [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+		return petMapper.EntityToUserDTO(petDao.save(petDao.getById(petDTO.getId())));				 
+	}
+
+	public PetDTO getPet(Long petId) {
+
+		return petMapper.EntityToUserDTO(petDao.getById(petId));
+	}
+
+	@Override
+	public PetDTO deletePet(PetDTO petDTO) {
```
Comment (luisod78) [inline DiegoFlores/SpringCRUD/src/main/java/com/bootcamp/vetPatitas/service/PetServiceImp.java:66]: Do you really need a DTO to delete a record??
only the id would be sufficient ?

File: DiegoFlores/SpringCRUD/src/main/java/com/bootcamp/vetPatitas/service/PetServiceImp.java [hunk 12]
```diff
@@ -0,0 +1,74 @@
...
+
+		PetDTO petToBeDesabled = new PetDTO();
```
Comment (luisod78) [inline DiegoFlores/SpringCRUD/src/main/java/com/bootcamp/vetPatitas/service/PetServiceImp.java:68]: typo: desabled ?

File: DiegoFlores/SpringCRUD/src/main/java/com/bootcamp/vetPatitas/exception/InvalidDataException.java [hunk 13]
```diff
@@ -0,0 +1,11 @@
+package com.bootcamp.vetPatitas.exception;
+
+public class InvalidDataException  extends NullPointerException {
```
Comment (luisod78) [inline DiegoFlores/SpringCRUD/src/main/java/com/bootcamp/vetPatitas/exception/InvalidDataException.java:3]: extends from NullPointerException???

you throw this Exception when input data is not correct... not when is null... right??

**Review by luisod78 — CHANGES_REQUESTED** [review]

**Review by Dfloresa — COMMENTED** [review]
The corrections were done

**Review by luisalanmz — APPROVED** [review]
LGTM!

**Pull request merged**